### PR TITLE
Modernize cmake: Prefer using targets. Fix CMake warnings

### PR DESCRIPTION
**Pull request by phcerdan**
# PR Description

Prefer using targets for Eigen and ZLIB:
- Require at least Eigen3.3 to use Eigen3Config.cmake and Eigen3::Eigen
target. Remove unused file: FindEigen3.cmake after this change. 
- Use target ZLIB::ZLIB instead of target_include_directories and
libraries.
- Also, because Boost and Zlib are required dependencies, remove the
`if(Boost_FOUND)` conditional.

CMake: Fix policy warning CMP0048 about project not handling VERSION:

- CMake does not allow yet not numeric versions, (DGtal use `beta` in version)
so we just suppress the policy warning setting it to OLD behavior.

CMake: Remove linking with DGtalLibDependencies in examples:

- DGtalLibDependencies is a list of libraries, but not needed since
recent modernization of CMake files. The target `DGtal` contains those libraries by itself.

# Checklist

- [NA] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [NA] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

**Comment by phcerdan** [comment]
This solves the failing travis-ci with doxygen Fix #1542

File: ChangeLog.md [hunk 1]
```diff
@@ -7,6 +7,8 @@
     (Pablo Hernandez-Cerdan [#1529](https://github.com/DGtal-team/DGtal/pull/1529))
   - Modernize CMake: Avoid global includes and links, use `target_` commands instead
     (Pablo Hernandez-Cerdan, David Coeurjolly [#1524](https://github.com/DGtal-team/DGtal/pull/1524))
+  - Modernize CMake: Prefer use targets than directories and libraries
```
Comment (dcoeurjo) [inline ChangeLog.md:10]:
```suggestion
  - Modernize CMake: Prefer use targets rather than directories and libraries
```

**Review by dcoeurjo — CHANGES_REQUESTED** [review]
Fine for me, thanks @phcerdan ! Just a minor comment for the changelog

File: ChangeLog.md (hunk 1 above)
Comment (phcerdan) [reply ChangeLog.md]: Thanks, fixed. I also removed linking to the list of libraries `${DGtalLibDependencies}` in examples. Those libraries are in the DGtal target already.

**Comment by dcoeurjo** [comment]
Perfect. Merging.
I would need to fix the pending azure hook..

**Review by dcoeurjo — APPROVED** [review]

**Pull request merged**